detect_regressions.py

```diff
@@ -0,0 +1,121 @@
+                    "avg_memory_last-2": df_passed.average_memory.iloc[-3],
+                    "peak_memory_mean": df_passed.peak_memory[-13:-3].mean(),
+                    "peak_memory_std": df_passed.peak_memory[-13:-3].std(),
+                    "peak_memory_last": df_passed.peak_memory.iloc[-1],
+                    "peak_memory_last-1": df_passed.peak_memory.iloc[-2],
+                    "peak_memory_last-2": df_passed.peak_memory.iloc[-3],
+                }
+
+                stats = stats_dict[f"({runtime, name})"]
+                dur_threshold = stats["duration_mean"] + stats["duration_std"]
+                avg_mem_threshold = stats["avg_memory_mean"] + stats["avg_memory_mean"]
+                peak_mem_threshold = (
+                    stats["peak_memory_mean"] + stats["peak_memory_mean"]
+                )
+
+                # Only raise if the last three show regression
+                if (
+                    stats["duration_last"] >= dur_threshold
+                    and stats["duration_last-1"] >= dur_threshold
+                    and stats["duration_last-2"] >= dur_threshold
+                ):
+                    reg = (
+                        f"{runtime= }, {name= }, "
+                        f"last_three_durations = "
+                        f"{(stats['duration_last'], stats['duration_last-1'],stats['duration_last-2'])}, "
+                        f"{dur_threshold= } \n"
+                    )
+
+                    regressions.append(reg)
+
+                if (
+                    stats["avg_memory_last"] >= avg_mem_threshold
+                    and stats["avg_memory_last-1"] >= avg_mem_threshold
+                    and stats["avg_memory_last-2"] >= avg_mem_threshold
+                ):
+                    reg = (
+                        f"{runtime= }, {name= }, "
+                        f"avg_mem_last = "
+                        f"{(stats['avg_memory_last'], stats['avg_memory_last-1'], stats['avg_memory_last-2'])}, "
+                        f"{avg_mem_threshold= } \n"
+                    )
+
+                    regressions.append(reg)
+
+                if (
+                    stats["peak_memory_last"] >= peak_mem_threshold
+                    and stats["peak_memory_last-1"] >= peak_mem_threshold
+                    and stats["peak_memory_last-2"] >= peak_mem_threshold
+                ):
+                    reg = (
+                        f"{runtime= }, {name= }, "
+                        f"peak_mem_last = "
+                        f"{(stats['peak_memory_last'], stats['peak_memory_last-1'], stats['peak_memory_last-2'])}, "
+                        f"{peak_mem_threshold= } \n"
+                    )
+
+                    regressions.append(reg)
+
+    # convert dict to dataframe to write to xml to use later ask ian if this is what he had in mind.
+    # df_stats = pandas.DataFrame.from_dict(stats_dict, orient="index")
+    # df_stats.to_xml("stats.xml")
```

Comment:
This is what I had in mind: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary

Looks like it's unstructured markdown, not XML

Reply:
So it seems that we can do `df.to_markdown()`. Now after seeing the example in the I assume that what you want in the summary is a summary of the regressions as a table, not all the stats for every test. 

Having a summary of the regressions as a nice markdown I think is possible. I'm thinking something like every row is a test_name, and as columns we have `[runtime_ver, regression_type, mean, last_val, last_val-1, last_val-2, thershold]` . 

Would something like this work for a summary?

Reply:
Sure, I don't have a strong idea of the best presentation, so if you come up with something that you think is useful, that would be a great starting point.